homeassistant/components/tibber/sensor.py

```diff
@@ -514,3 +514,89 @@ def get_live_measurement(self):
             _LOGGER.error(errors[0])
             return None
         return self.data.get("data", {}).get("liveMeasurement")
+
+
+class TibberDataCoordinator(update_coordinator.DataUpdateCoordinator):
+    """Handle Tibber data and insert statistics."""
+
+    def __init__(self, hass, tibber_connection):
+        """Initialize the data handler."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=f"Tibber {tibber_connection.name}",
+            update_interval=timedelta(hours=1),
+        )
+        self._tibber_connection = tibber_connection
+
+    async def _async_update_data(self):
+        """Update data via API."""
+        await self._tibber_connection.fetch_consumption_data_active_homes()
+        await self._insert_statistics()
+
+    async def _insert_statistics(self):
+        """Insert Tibber statistics."""
+        for home in self._tibber_connection.get_homes():
+            if not home.hourly_consumption_data:
+                continue
+
+            statistic_id = (
+                f"{TIBBER_DOMAIN}:{home.home_id.replace('-', '')}_energy_consumption"
+            )
+
+            last_stats = await self.hass.async_add_executor_job(
+                get_last_statistics, self.hass, 1, statistic_id, True
+            )
+
+            if not last_stats:
+                hourly_consumption_data = await home.get_historic_data(5 * 365 * 24)
```

Comment:
Can we explain what's going on here? Should we ask for a few more days to cover leap years, or Tibber only stored 5*365 days worth of data?

Reply:
I thought 5 years would be enough?  It will be limited to the time the user has been a Tibber customer.
Should I increase it?

Reply:
It's fine, I basically meant to add comments in the code 😅